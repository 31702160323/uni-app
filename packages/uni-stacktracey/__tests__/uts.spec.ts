import path from 'path'

const {
  generateCodeFrameWithAndroidStacktrace,
} = require('../dist/uni-stacktracey.cjs.js')

const utsProjectDir = path.resolve(__dirname, '../test/uts')

describe('code-frame-uts', () => {
  test('android', async () => {
    expect(
      await generateCodeFrameWithAndroidStacktrace(androidUniModulesError, {
        name: 'uni_modules/test-uts1',
        inputDir: '/Users/fxy/DCloud/test-uts',
        outputDir: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
      })
    ).toMatchSnapshot()
    expect(
      await generateCodeFrameWithAndroidStacktrace(androidUtssdkError, {
        name: 'utssdk/test2',
        inputDir: '/Users/fxy/DCloud/test-uts',
        outputDir: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
      })
    ).toMatchSnapshot()
  })
})

const androidUniModulesError = `e: uni_modules/test-uts1/utssdk/app-android/index.kt: (8, 12): The integer literal does not conform to the expected type String

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':uni_modules:test-uts1:compileReleaseKotlin'.
> Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 50s`

const androidUtssdkError = `e: utssdk/test2/app-android/index.kt: (8, 12): The integer literal does not conform to the expected type String

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':uni_modules:test-uts1:compileReleaseKotlin'.
> Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 50s`
