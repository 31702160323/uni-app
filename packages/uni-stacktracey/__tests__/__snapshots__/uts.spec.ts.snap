// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-frame-uts android 1`] = `
"error: The integer literal does not conform to the expected type String
at uni_modules/test-uts1/utssdk/index.uts:2:10
1  |  export function test(): string {
2  |    return 1;
   |            ^
3  |  }
4  |  


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':uni_modules:test-uts1:compileReleaseKotlin'.
> Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 50s"
`;

exports[`code-frame-uts android 2`] = `
"error: The integer literal does not conform to the expected type String
at utssdk/test2/index.uts:2:10
1  |  export function test(): string {
2  |    return 2;
   |            ^
3  |  }
4  |  


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':uni_modules:test-uts1:compileReleaseKotlin'.
> Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 50s"
`;
